### ISSUE TITLE
Allow going back on menu

### ISSUE THREAD

**Issue by stacksjb** [alanvardy/tod#729]
If I incorrectly select "Pick project" and really wanted to delete, there is no way to go back.

This should probably be a default universal thing, such as pressing left arrow when no text is present, pressing escape (currently exits the program instead of going back a menu), another keyboard shortcut, or simply a "Go back to previous" option added to all menus.

**Comment by stacksjb** [comment]
It appears this does not work on "Select Project", but it does work as expected on each task under "tod list process" so it may be inconsistent. 

I can go through and test each function(s) if needed.

**Comment by stacksjb** [comment]
I think using "escape" may be appropriate (I can always ctrl-c to break).

"Tod list process" also simply cancels if I hit escape and exits.

**Comment by alanvardy** [comment]
> I can go through and test each function(s) if needed.

This would be a big help if you can :pray:

**Comment by stacksjb** [comment]
I will look through this, thx.

**Comment by alanvardy** [comment]
I don't think that I can remap, but I can definitely match on the error call the function again so that the user can select from the beginning

```
"Operation was canceled by the user"
```

**Comment by alanvardy** [comment]
- [ ] project
	- [ ] remove -> No back/cancel menu option
	- [ ]   rename -> No back/cancel menu option
	- [ ]   import -> no back/cancel menu option
	- [ ]   empty -> no back cancel menu option
- [ ] task
	- [ ]   quick-add -> Single command, though no way to cancel/go back
	- [ ]   create -> single command, no way to cancel/go back
	- [ ]   edit -> no back/cancel menu option
	- [ ]   next -> no back/cancel menu option
	- [ ]   complete -> single command ✅
- [ ] list
	- [ ]   view -> No back/cancel from project/filter selection option
	- [ ]   process -> no back/cancel from project/filter, but the end options do have a “Quit” (I think this could be remapped ot back instead - if I want to quit I can always just ctrl-c or close the terminal window)
	- [ ]   label -> no back/cancel once labels are entered (in seleciton menu)
	- [ ]   schedule -> no back/cancel option

**Comment by stacksjb** [comment]
I ran into this again recently on teh schedule option - when selecting date, no "q" or "esc" option listed. 

So linked with this would be a need to add documentation to the prompt, i.e.

`[arrows to move, []{} move months and years, enter to select, esc to cancel]`

**Comment by alanvardy** [comment]
^ Split out into a different issue

**Comment by stacksjb** [comment]
I've still run into this a few times, but understanding the code I'm not sure how easily it can be addressed.

Essentially the workflow is that if I'm working through a list, and I skip something by accident, there's no easy way to go back without losing my progress, nor to restart without going through everything again.

But fixing this might require some sort of tracking implementation (probably overkill), and we can easily just remind people that part of the point is to implement the "[one thing at a time mentality](https://jamesclear.com/master-one-thing)"

So, as long as we can consistently exist/end/skip I think I'm happy (if I have to cntrl-C etc to kill the program them I feel like something is broken).

**Comment by stacksjb** [comment]
Just thinking out loud here, but maybe there is a default option you can define in code somewhere and then just reuse on all functions? Although if it's just the one that may not really save you much effort.

**Comment by stacksjb** [comment]
Note to self:

every Inquire prompt (Select, Text, DateSelect, MultiSelect, etc.) returns a Result<T, InquireError>. There are two distinct "user backed out" variants:

InquireError::OperationCanceled — user pressed Esc
InquireError::OperationInterrupted — user pressed Ctrl-C

Right now OperationCanceled isn't handled, so it's treated the same. Need to explicitly match on that, then add a loop for the main menu.
